### DISCUSSION TITLE
Rewrite concern with 3500+ entries in my next.config.js

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/next.js#28566] [Help]
I'm converting a large site (current site converted is at https://svcc.mobi/presenter/2019/douglas-crockford-1124), and unfortunately, I've been not consistent over the past 15 years with casing in the URL's.  Sometimes I use:

https://svcc.mobi/presenter/2019/douglas-crockford-1124 and other times, make the p in presenter P.  https://svcc.mobi/Presenter/2019/douglas-crockford-1124.  That goes for many other landing pages also.  I machine generated my `next.config.js` to include all the re-writes to work, but I get a warning saying over 1000 will cause performance problems.

My question is, 1) Where are the performance degrades so I can see if they are an issue for me, and 2) Is there any kind of wild-card like way I can solve the issue for all URL's?  That is. Any URL that comes in with svcc.mobi/Presen... gets converted to svcc.mobi/presen... with only changing the P and leaving everything before and after it the same.  

I'm a regex beginner so not sure if that's possible.

Below is a sample of what my next.config.js file looks like.

```
...
  { source: '/(n|N)ews', destination: '/news' },
  { source: '/(n|N)ews/2017', destination: '/news/2017' },
  { source: '/(n|N)ews', destination: '/news' },
  { source: '/(n|N)ews/2018', destination: '/news/2018' },
  { source: '/(n|N)ews', destination: '/news' },
  { source: '/(n|N)ews/2019', destination: '/news/2019' },
  { source: '/(a|A)bout', destination: '/about' },
  { source: '/(P|p)ast(E|e)vents', destination: '/pastevents' },
  { source: '/(c|C)ode(o|O)f(c|C)onduct', destination: '/codeofconduct' },
  { source: '/(V|v)ideo(S|s)vcc', destination: '/videosvcc' },
  { source: '/(F|f)(A|a)(Q|q)', destination: '/faq' },
  { source: '/(p|P)resenter', destination: '/presenter' },
  { source: '/(p|P)resenter/2006', destination: '/presenter/2006' },
  { source: '/(p|P)resenter', destination: '/presenter' },
  { source: '/(p|P)resenter/2007', destination: '/presenter/2007' },
  { source: '/(p|P)resenter', destination: '/presenter' },
  {
    source:
      '/(s|S)ession/2008/an-introduction-to-comet-and-bayeux-pushing-data-to-the-browser',
    destination:
      '/session/2008/an-introduction-to-comet-and-bayeux-pushing-data-to-the-browser',
  },
  {
    source: '/(s|S)ession/2008/java2d-and-groovy-a-perfect-match',
    destination: '/session/2008/java2d-and-groovy-a-perfect-match',
  },
  {
    source: '/(s|S)ession/2008/boosting-your-testing-productivity-with-groovy',
    destination: '/session/2008/boosting-your-testing-productivity-with-groovy',
  },
  {
    source:
      '/(s|S)ession/2008/building-rich-applications-with-groovys-swingbuilder',
    destination:
      '/session/2008/building-rich-applications-with-groovys-swingbuilder',
  },
...
```

**Comment by samcx** [comment]
Unfortunately, this is not possible via regex because there is no value we can check in the destination. This is something we should definitely support!

**Comment by styfle** [comment]
You probably want to use `redirects`, not `rewrites`, so you can avoid duplicate content.

This part:

```
redirects: [
  { source: '/(p|P)resenter', destination: '/presenter' },
  { source: '/(p|P)resenter/2006', destination: '/presenter/2006' },
  { source: '/(p|P)resenter', destination: '/presenter' },
  { source: '/(p|P)resenter/2007', destination: '/presenter/2007' },
  { source: '/(p|P)resenter', destination: '/presenter' },
]
```

can be simplified to this:

```
redirects: [
  { source: '/Presenter/:path*', destination: '/presenter/:path*' },
]
```

Learn more: https://nextjs.org/docs/api-reference/next.config.js/redirects
